### PR TITLE
fix: Improve task & event logging

### PR DESCRIPTION
**Pull request by karlkfi**
- Log tasks at start and end
- Log events before sending
- Add String funcs for readable event logs

### CODE REVIEW

**Review by seans3 — APPROVED** [review]

File: pkg/apply/taskrunner/context.go [hunk 1]
```diff
@@ -53,6 +53,12 @@ func (tc *TaskContext) ResourceCache() cache.ResourceCache {
 	return tc.resourceCache
 }
 
+// SendEvent sends an event on the event channel
+func (tc *TaskContext) SendEvent(e event.Event) {
+	klog.V(5).Infof("sending event: %s", e)
```
Comment (ash2k) [inline pkg/apply/taskrunner/context.go:58]: I'd make this debug level. Events are usually consumed by printers, so there is no need to log it and then also print it.

File: pkg/apply/taskrunner/context.go [hunk 2]
```diff
@@ -53,6 +53,12 @@ func (tc *TaskContext) ResourceCache() cache.ResourceCache {
 	return tc.resourceCache
 }
 
+// SendEvent sends an event on the event channel
+func (tc *TaskContext) SendEvent(e event.Event) {
```
Comment (ash2k) [inline pkg/apply/taskrunner/context.go:57]: I like this new method, it's a good idea. I wonder if we can delete the `EventChannel()` method to encapsulate the channel even more? No need to do it in this PR, of course.

Comment (karlkfi) [reply pkg/apply/taskrunner/context.go:57]: The EventChannel method is used in a few places to read from the event channel, like in the ApplyTask which sends it to `applyOptionsFactoryFunc` (aka `newApplyOptions`) which passes it to `KubectlPrinterAdapter`. I don't really understand why it's building a printer, but if you wanna clean it up, be my guest.

I mostly just added the SendEvent convenience method so there was a single place to log from.

File: pkg/apply/taskrunner/context.go (hunk 1 above)
Comment (karlkfi) [reply pkg/apply/taskrunner/context.go:58]: AFAICT cli-utils has no debug or even warning logging. It's all info and error logs with various levels of verbosity.

It's not really clear to me when to use each type with klog or even really what the levels are intended for. But if we want to standardize it should probably be a different cleanup PR.

**Pull request merged**